activity_calendar/templatetags/calendar_tags.py

```diff
@@ -0,0 +1,82 @@
+import datetime
+
+from django import template
+from django.utils.formats import date_format
+
+from activity_calendar.forms import RegisterForActivitySlotForm
+
+register = template.Library()
+
+@register.simple_tag(takes_context=True)
+def sign_up_slot_form(context, slot):
+    user = context['request'].user
+
+    # There is no logged in user, so return nothing instead
+    if not user.is_authenticated:
+        return None
+
+    form = RegisterForActivitySlotForm(
+        activity = context['activity'],
+        user = user,
+        recurrence_id=context['recurrence_id'],
+        slot = slot,
+        initial = {
+            'slot_id': slot.id,
+            'sign_up': slot.get_subscribed_participants().filter(id=user.id).count() == 0
+        }
+    )
+    return form
+
+
+@register.simple_tag(takes_context=True)
+def get_opening_time(context, filter=None):
```

Comment:
Also displays for past events whose subscriptions actually won't ever open again.